### ISSUE TITLE
Failed to install restheart docker on AWS elasticbeanstalk

### ISSUE THREAD

**Issue by daipingzhao** [SoftInstigate/restheart-docker#6]
I just configure my own mongodb url and zip it as a package, then upload it to AWS elasticbeanstalk. But unfortunately, it always failed. I am using elasticbeanstalk docker model as the link: http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/create_deploy_docker.html

Here is some error log sanpshot form aws server:
----------------------------------------------------------------------------
Step 1 : FROM anapsix/alpine-java:jre8
---> 157361c9e6a0
Step 2 : MAINTAINER SoftInstigate maurizio@softinstigate.com
---> Using cache
---> bddebefce499
Step 3 : ENV release 1.1.7
---> Using cache
---> 24402459d1e9
Step 4 : RUN apk update && apk upgrade && apk add curl
---> Using cache
---> e747ff6a7ae4
Step 5 : WORKDIR /opt/
---> Using cache
---> afa3108c2fb4
Step 6 : COPY nexus.sh /opt/
---> Using cache
---> 8da695c48512
Step 7 : RUN ./nexus.sh -i org.restheart:restheart:${release} -p tar.gz > restheart.tar.gz && tar -zxvf restheart.tar.gz && mv restheart-${release} restheart && rm -f restheart.tar.gz
---> Running in 94af1a9dd4e4
[91m/bin/sh: ./nexus.sh: Permission denied
[0mThe command '/bin/sh -c ./nexus.sh -i org.restheart:restheart:${release} -p tar.gz > restheart.tar.gz && tar -zxvf restheart.tar.gz && mv restheart-${release} restheart && rm -f restheart.tar.gz' returned a non-zero code: 126
Failed to build Docker image aws_beanstalk/staging-app: tar.gz > restheart.tar.gz && tar -zxvf restheart.tar.gz && mv restheart-${release} restheart && rm -f restheart.tar.gz' returned a non-zero code: 126.

**Comment by mkjsix** [comment]
Is there a way you can execute the `docker pull softinstigate/restheart` command and download the public image instead of building it from scratch? The source of your error seems to be `./nexus.sh: Permission denied` but I have still no clue of why it happens in your case, as it's just trying to download the restheart.tgz from Maven Central.

**Comment by daipingzhao** [comment]
I can connect to the ec2 server which host my  elasticbeanstalk docker. Then I can use commands
  [ec2-user@ip-172-31-42-142 ~]$ sudo docker pull softinstigate/restheart

 It can download it correctly, no issue, here the result output

latest: Pulling from softinstigate/restheart
Digest: sha256:9fa9fa9f75dc3d2056eb17e83a3c6bbe29876240793a29093804f0b9f06812e8
Status: Image is up to date for softinstigate/restheart:latest

**Comment by mkjsix** [comment]
So if you can avoid building the image and just use the official one you have pulled, then the issue should disappear

**Comment by daipingzhao** [comment]
Actually, I want to use elasticbeanstalk docker feature, which means that I can use AWS managed service. But AWS elasticbeanstalk docker features only accepts dockfile, If I do it manually, then it is another story.

**Comment by daipingzhao** [comment]
We contacted AWS support and they said it should be restheart issue. Can you make resheart compatible with aws beanstalk?"

**Comment by mkjsix** [comment]
So far I don't have a clue on what I should change, as here the build is working perfectly, as you can see from the automated build logs: https://hub.docker.com/r/softinstigate/restheart/builds/bqx9c5dyqggpdxmctu7h4yp/

I need some more information on what is exactly failing on ELB. The return code 126 usually means "command not executable", but I can't reproduce the issue neither locally nor on the Docker hub.

I suggest you to edit the Dockerfile so that the line

```
RUN ./nexus.sh -i org.restheart:restheart:${release} -p tar.gz > restheart.tar.gz \
&& tar -zxvf restheart.tar.gz \
&& mv restheart-${release} restheart \
&& rm -f restheart.tar.gz
```

Becomes:

```
RUN ./nexus.sh -i org.restheart:restheart:${release} -p tar.gz > restheart.tar.gz 
RUN tar -zxvf restheart.tar.gz
RUN mv restheart-${release} restheart \
RUN rm -f restheart.tar.gz
```

So that we could try to identify which is the offending command in your case.

**Comment by mkjsix** [comment]
Besides, I think you should ask to AWS support why this Dockerfile is working everywhere except on ELB. If the Dockerfile had problems then it should be broken everywhere, but this is not the case...

**Comment by mkjsix** [comment]
We have just released RESTHeart 2.0.0 final. The Dockerfile has been updated to download from Github instead of Maven Central.
